packages/frontend/electron/src/main/updater/index.ts

```diff
@@ -1,7 +1,51 @@
-import { app } from 'electron';
+import { app, type IpcMainInvokeEvent } from 'electron';
 
 import type { NamespaceHandlers } from '../type';
-import { checkForUpdates, quitAndInstall } from './electron-updater';
+import {
+  checkForUpdates,
+  downloadUpdate,
+  getAllowAutoCheckUpdate,
+  getAllowAutoDownloadUpdate,
+  quitAndInstall,
+  setAllowAutoCheckUpdate,
+  setAllowAutoDownloadUpdate,
+} from './electron-updater';
+
+export type UpdaterConfig = {
+  autoCheckUpdate: boolean;
+  autoDownloadUpdate: boolean;
+};
+
+const config: UpdaterConfig = {
+  autoCheckUpdate: getAllowAutoCheckUpdate(),
+  autoDownloadUpdate: getAllowAutoDownloadUpdate(),
+};
```

Comment:
Perhaps you can just remove `getAllowAutoCheckUpdate`,  `getAllowAutoDownloadUpdate` etc in `./electron-updater` and manipulate the updater config directly there, instead of breaking down them into individuals and then use a proxy to get the latest value. It is a bit overkill in this simple case.